src/averageAges.js

```diff
@@ -55,10 +84,24 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const childrenWithMothers = people.filter(child => {
+    return onlyWithSon
+      ? people.find(woman =>
+        child.sex === sexMale && child.mother === woman.name)
+      : people.find(woman => child.mother === woman.name);
+  });
+
+  const ageDiffBetween = childrenWithMothers.map(child => {
+    const mother = people.find(person => person.name === child.mother);
+
+    return child.born - mother.born;
+  });
+
+  const sumAgeDiffBetween = ageDiffBetween.reduce((sum, ageDiff) =>
+    sum + ageDiff, 0);
```

Comment:
Move the logic from the `map` to the middle of the `reduce` so as not to go through the arrays one more time. 
Also, don't forget to use destructuring for the object.